temporal-sdk/src/main/java/io/temporal/worker/tuning/ResourceBasedTuner.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.worker.tuning;
+
+import io.temporal.common.Experimental;
+import java.time.Duration;
+import javax.annotation.Nonnull;
+
+/** A {@link WorkerTuner} that attempts to allocate slots based on available system resources. */
+@Experimental
+public class ResourceBasedTuner implements WorkerTuner {
+  public static final ResourceBasedSlotOptions DEFAULT_WORKFLOW_SLOT_OPTIONS =
+      new ResourceBasedSlotOptions(2, 500, Duration.ZERO);
+  public static final ResourceBasedSlotOptions DEFAULT_ACTIVITY_SLOT_OPTIONS =
+      new ResourceBasedSlotOptions(1, 1000, Duration.ofMillis(50));
+
+  private final ResourceBasedController controller;
+  private final ResourceBasedSlotOptions workflowSlotOptions;
+  private final ResourceBasedSlotOptions activitySlotOptions;
+  private final ResourceBasedSlotOptions localActivitySlotOptions;
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
+    private ResourceBasedControllerOptions controllerOptions;
+    private @Nonnull ResourceBasedSlotOptions workflowSlotOptions = DEFAULT_WORKFLOW_SLOT_OPTIONS;
+    private @Nonnull ResourceBasedSlotOptions activitySlotOptions = DEFAULT_ACTIVITY_SLOT_OPTIONS;
+    private @Nonnull ResourceBasedSlotOptions localActivitySlotOptions =
+        DEFAULT_ACTIVITY_SLOT_OPTIONS;
+
+    private Builder() {}
+
+    public Builder setControllerOptions(ResourceBasedControllerOptions controllerOptions) {
+      this.controllerOptions = controllerOptions;
+      return this;
+    }
+
+    /**
+     * Set the slot options for workflow tasks. Has no effect after the worker using this tuner
+     * starts.
+     *
+     * <p>Defaults to minimum 2 slots, maximum 500 slots, and no ramp throttle.
```

Comment:
These defaults don't align with the default poller count which is 5

Reply:
Changed to 5